onnxruntime/core/providers/shared_library/provider_api.h

```diff
@@ -129,15 +153,17 @@ class TensorShape : private std::vector<int64_t> {
   int64_t SizeHelper(size_t start, size_t end) const;
 };
 
-constexpr const char* kMSDomain = "com.microsoft";
-constexpr const char* kMklDnnExecutionProvider = "MKLDNNExecutionProvider";
-
 template <typename T>
 using IAllocatorUniquePtr = std::unique_ptr<T, std::function<void(T*)>>;
 
-std::unique_ptr<Provider_IDeviceAllocator> CreateCPUAllocator(std::unique_ptr<Provider_OrtMemoryInfo> memory_info);
-Provider_AllocatorPtr CreateAllocator(const Provider_DeviceAllocatorRegistrationInfo& info, int16_t device_id = 0,
-                                      bool use_arena = true);
+std::unique_ptr<Provider_IDeviceAllocator> Provider_CreateCPUAllocator(std::unique_ptr<Provider_OrtMemoryInfo> memory_info);
+std::unique_ptr<Provider_IDeviceAllocator> Provider_CreateCUDAAllocator(int16_t device_id, const char* name);
+std::unique_ptr<Provider_IDeviceAllocator> Provider_CreateCUDAPinnedAllocator(int16_t device_id, const char* name);
+Provider_AllocatorPtr CreateAllocator(const Provider_DeviceAllocatorRegistrationInfo& info, int16_t device_id = 0, bool use_arena = true);
```

Comment:
This file seem to contain common provider API, why provider specific things are here?

Reply:
This is the file that the providers include, but the core onnxruntime code doesn't include (the parts that both include are to implement the shared interface).